### ISSUE TITLE
Federation: Google Cloud DNS provider error 404 caused by the ProjectID

### ISSUE THREAD

**Issue by walteraa** [kubernetes/kubernetes#52726]
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**: I've tried to use Google DNS as DNS provider in a federation with FCP in an on-premises cluster. I've configured a secret with the Google credential file with correct permissions, I mounted the secret as a volume and assigned the "volume path" to the environment variable `GOOGLE_APPLICATION_CREDENTIALS`(before that I was receiving an authorization error, but this configuration fixed it) by editing the `federation-controller-manager` deployment. 

The credential file was generated from the [Google Cloud console credential page](https://console.developers.google.com/apis/credentials?pli=1&project=corc-tutorial) and has the following structure

```json
{
  "type": "service_account",
  "project_id": "corc-tutorial",
  "private_key_id": "{THE PRIVATE KEY ID}",
  "private_key": "-----BEGIN PRIVATE KEY-----\n
{THE PRIVATE KEY CONTENT}
\n-----END PRIVATE KEY-----\n",
  "client_email": "{ACCOUNT}",
  "client_id": "{CLIENT_ID}",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://accounts.google.com/o/oauth2/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/{GSERVICEACCOUNT_URI}"
}
```

Then I've got the following logs in the `federation-controller-manager` pod
```
I0822 13:17:33.998181       1 clouddns.go:100] Successfully got DNS service: &{0xc420e19b00 https://www.googleapis.com/dns/v1/projects/  0xc42000e840 0xc42000e848 0xc42000e850 0xc42000e858}
E0822 13:17:35.621941       1 dns.go:97] Failed to retrieve DNS zone: error querying for DNS zones: googleapi: got HTTP response code 404 with body: Not Found
F0822 13:17:35.622009       1 controllermanager.go:144] Failed to start service dns controller: error querying for DNS zones: googleapi: got HTTP response code 404 with body: Not Found
```

Analyzing the issue, I figured out that the Google cloud provider isn't getting the `project_id` key/value, which is causing a bug in the `ProjectID` property, thus generating an invalid URL when trying to get information from my DNS zone.

**What you expected to happen**: Once generated the credential file and configure it in the `federation-controller-manager` deployment as previously described, the Google DNS provider should get the DNS information from the REST API and successfully manage it.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create an on-premises cluster
2. Create a DNS zone in your Google cloud console
3. Initialize a federation which the FCP is the on-premises cluster created before and using Google as DNS provider and the DNS zone created before
4. Create a credential file giving permission to manage the project's DNS
5. Create a secret using the credential file created before
6. Edit the  `federation-controller-manager` by creating the volumes using the secret created before and an environment variable named `GOOGLE_APPLICATION_CREDENTIALS` by using secret's mount path as value.

**Anything else we need to know?**:
I did it work, as it should, doing a workaround which sets the `ProjectID` value hardcoded as you can see [here](https://github.com/walteraa/kubernetes/commit/3110b2cad1751eab499717c1f5f029bbb59f349e#diff-9e1208e1d901d951628091e4e80c06dcR58)

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.4
- Cloud provider or hardware configuration**: *irrelevant*
- OS (e.g. from /etc/os-release): *irrelevant*
- Kernel (e.g. `uname -a`): *irrelevant*
- Install tools: *irrelevant*
- Others: *irrelevant*

**Comment by walteraa** [comment]
Closed because in hyperkube tagged as `v1.8.0`, the Google DNS is working with the same conditions that I described in the issue.

**Comment by irfanurrehman** [comment]
This issue was labelled only for sig/multicluster and is thus moved over to kubernetes/federation#10.
If this does not seem to be right, please reopen this and notify us @kubernetes/sig-multicluster-misc.
/close
